stdlib/public/core/Arrays.swift.gyb

```diff
@@ -2181,37 +2162,81 @@ extension ${Self} {
 %end
 
 #if _runtime(_ObjC)
+import SwiftShims
+
+internal enum _ArrayBridgeFailure : Error {
+case count(Int)
+}
+
+extension _ContiguousArrayBuffer {
+  internal init?(_bridgingFromObjectiveC source: AnyObject) {
+    typealias Storage = _ContiguousArrayStorage<Element>
+    
+    // First see if there's already a native buffer we can grab
+    if let storage = (source as? Storage)
+      ?? (source as? _SwiftDeferredNSArray)?._nativeStorage as? Storage {
+      self.init(storage)
+    }
+    else {
+      self.init(_slowBridgingFromObjectiveC: source)
+    }
+  }
+
+  internal init?(_slowBridgingFromObjectiveC source: AnyObject) {
+    let nsArray = unsafeBitCast(source, to: _NSArrayCore.self)
+    let count = nsArray.count
+    self.init(_uninitializedCount: count)
+    if count == 0 { return }
+
+    if (_isBridgedVerbatimToObjectiveC(Element.self)) {
+      withUnsafeMutableBufferPointer {
+        $0.baseAddress!.withMemoryRebound(to: AnyObject.self, capacity: count) {
+          nsArray.getObjects($0, range: _SwiftNSRange(location: 0, length: count))
```

Comment:
so the AUMP case was explicitly removed from the swift interface of NSArray since it actually does not represent the API contract that NSArray has.

`- (void)getObjects:(ObjectType __unsafe_unretained [])objects range:(NSRange)range NS_SWIFT_UNAVAILABLE("Use 'subarrayWithRange()' instead");`

This API should not be imported as AutoreleasingUnsafeMutablePointer because NSArray does not autorelease those objects: they are actually passed as Unmanaged.